packages/storage-ui/src/Contexts/StorageApiContext.tsx

```diff
@@ -226,99 +215,35 @@ const StorageApiProvider = ({ apiUrl, withLocalStorage = true, children }: Stora
-      setReturningUser()
-      return Promise.resolve()
-    } catch {
-      return Promise.reject("There was an error logging in")
+    if (!isReady) {
+      const connected = await checkIsReady()
+      if (!connected) return Promise.reject("You need to allow the connection")
     }
-  }
+    const signer = provider.getSigner()
+    const address = await signer.getAddress()
 
-  const loginWithGoogle = async (
-    code: string,
-    state: string,
-    scope: string | undefined,
-    authUser: string | undefined,
-    hd: string | undefined,
-    prompt: string | undefined
-  ) => {
     try {
-      const {
-        access_token,
-        refresh_token
-      } = await storageApiClient.postOauth2CodeGoogle(
-        code,
-        state,
-        scope,
-        authUser,
-        hd,
-        prompt
-      )
+      const { token: welcomeMessage } = await storageApiClient.getIdentityWeb3Token(address)
 
-      setTokensAndSave(access_token, refresh_token)
-      setReturningUser()
-      return Promise.resolve()
-    } catch (err) {
-      return Promise.reject("There was an error logging in")
-    }
-  }
+      const signature = await signer.signMessage(welcomeMessage)
+      const { token: providerIdentityToken } = await storageApiClient.postIdentityWeb3Token({
+        signature: signature,
+        token: welcomeMessage,
+        public_address: address
+      })
 
-  const loginWithFacebook = async (code: string, state: string) => {
-    try {
-      const {
-        access_token,
-        refresh_token
-      } = await storageApiClient.postOauth2CodeFacebook(code, state)
+      const { access_token, refresh_token } = await storageApiClient.loginUser({
+        provider: "web3",
+        service: "storage",
+        token: providerIdentityToken
+      })
 
       setTokensAndSave(access_token, refresh_token)
       setReturningUser()
       return Promise.resolve()
-    } catch (err) {
-      return Promise.reject("There was an error logging in")
-    }
-  }
-
-  const web3Login = async () => {
-    if (!provider) return Promise.reject("No wallet is selected")
-
-    if (!isReady) {
-      const connected = await checkIsReady()
-      if (!connected) return Promise.reject("You need to allow the connection")
-    }
-    const signer = provider.getSigner()
-    try {
-      const { token } = await storageApiClient.getWeb3Token()
-
-      if (token) {
-        const signature = await signer.signMessage(token)
-        const address = await signer.getAddress()
-        const {
-          access_token,
-          refresh_token
-        } = await storageApiClient.postWeb3Token({
-          signature: signature,
-          token: token,
-          public_address: address
-        })
-        setTokensAndSave(access_token, refresh_token)
-        setReturningUser()
-        return Promise.resolve()
-      }
     } catch (error) {
       return Promise.reject("There was an error logging in.")
```

Comment:
In case it helps future us debug.
```suggestion
      console.error(error)
      return Promise.reject("There was an error logging in.")
```